pkg/auth/dependency/interaction/adaptors/identity.go

```diff
@@ -72,6 +73,24 @@ func (a *IdentityAdaptor) GetByClaims(typ authn.IdentityType, claims map[string]
 			return "", nil, err
 		}
 		return o.UserID, oauthToIdentityInfo(o), nil
+
+	case "":
+		// return any identity with the specified claims
+		userID, i, err := a.GetByClaims(authn.IdentityTypeLoginID, claims)
+		if err != nil && !errors.Is(err, identity.ErrIdentityNotFound) {
+			return "", nil, err
+		} else if err == nil {
+			return userID, i, nil
+		}
+
+		userID, i, err = a.GetByClaims(authn.IdentityTypeOAuth, claims)
+		if err != nil && !errors.Is(err, identity.ErrIdentityNotFound) {
+			return "", nil, err
+		} else if err == nil {
+			return userID, i, nil
+		}
+
+		return "", nil, identity.ErrIdentityNotFound
```

Comment:
Since it is used for checking for duplicated identity, I think instead I'll add a `ExistsByClaims` function that returns a boolean indicating whether an identity with the claims exists.